### ISSUE TITLE
[BUG] Generated union copy constructor can be invalid

### ISSUE THREAD

**Issue by KendallHarter** [hsutter/cppfront#772]
**Describe the bug**
Copying a union can result in invalid pointers errors.

**To Reproduce**
Sample code:
```
test: @union type = {
   name: std::string;
}

main: () = {
   a1: test = ();
   a1.set_name("hi");
   a2 := a1;
}
```

Commands:
```
./cppfront test.cpp2
g++11.3 -I ./include/ -std=c++20 test.cpp
./a.out
```

Expected result:
Program runs without error.

Actual result/error:
```
munmap_chunk(): invalid pointer
Aborted
```

**Additional context**
This seems to be caused by the implicit member-wise copying having no opt-out option, as it'll copy the storage before potentially deleting it with the `set_name` function.

**Comment by JohelEGP** [comment]
Yes.
It copies the storage and discriminator.
So it believes it has a live object when the `set_*` first calls `destroy()`.

**Comment by JohelEGP** [comment]
Smells like #475.

**Comment by JohelEGP** [comment]
The general `operator=` should default-initialize the storage and set the discriminator to `-1`.

**Comment by JohelEGP** [comment]
Resolved by commit cdf71bdca64536a005f2491d8c19f1d05a1c62f6.
